File: 0001-0500/0080.Remove Duplicates from Sorted Array II.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i, j = 0, 0
        while j < len(nums):
            if i < 2 or nums[i - 2] != nums[j]:
                nums[i] = nums[j]
                i += 1
            j += 1
        return i

File: 0001-0500/0080.Remove Duplicates from Sorted Array II.java
/**
 *Basic Idea:
 * compare nums[pioneer] with nums[cnt-1], rather than nums[cnt].
 *Result:
 * 164 / 164 test cases passed.
 * Status: Accepted
 * Runtime: 1 ms
 * Your runtime beats 59.44% of javasubmissions.
 *Date:
 * 9/1/2016
 */
public class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums==null || nums.length==0)
            return 0;
        if(nums.length==1)
            return 1;
        int cnt = 1;
        int pioneer = 2;
        while(pioneer<nums.length){
            if(nums[cnt-1]!=nums[pioneer]){
                nums[++cnt]=nums[pioneer];
            }
            pioneer++;
        }
        return cnt+1;
    }
}
